apps/extension/src/state/connected-sites.ts

```diff
@@ -2,74 +2,41 @@ import { ExtensionStorage } from '@penumbra-zone/storage/src/chrome/base';
 import { LocalStorageState, OriginRecord } from '@penumbra-zone/storage/src/chrome/types';
 import { AllSlices, SliceCreator } from '.';
 
-import Map from '@penumbra-zone/polyfills/src/Map.groupBy';
-import { UserChoice } from '@penumbra-zone/types/src/user-choice';
-
 export interface ConnectedSitesSlice {
   filter?: string;
-  knownSites: OriginRecord[];
-  approvedSites: OriginRecord[];
-  deniedSites: OriginRecord[];
-  ignoredSites: OriginRecord[];
-  noFilterMatch?: boolean;
-  loadKnownSites: () => Promise<void>;
   setFilter: (search?: string) => void;
+  knownSites: OriginRecord[];
   discardKnownSite: (originRecord: OriginRecord) => Promise<void>;
+  frontendUrl?: string;
+  setFrontendUrl: (frontendUrl: string) => void;
 }
 
 export const createConnectedSitesSlice =
   (local: ExtensionStorage<LocalStorageState>): SliceCreator<ConnectedSitesSlice> =>
   (set, get) => ({
-    filter: undefined,
-    noFilterMatch: undefined,
     knownSites: [],
-    approvedSites: [],
-    deniedSites: [],
-    ignoredSites: [],
-
-    loadKnownSites: async () => {
-      const knownSites = await local.get('knownSites');
-      const groupedSites = Map.groupBy(knownSites, ({ choice }) => choice);
 
+    filter: undefined,
+    setFilter: (search?: string) => {
       set(state => {
-        state.connectedSites.knownSites = knownSites;
-        state.connectedSites.approvedSites = groupedSites.get(UserChoice.Approved) ?? [];
-        state.connectedSites.deniedSites = groupedSites.get(UserChoice.Denied) ?? [];
-        state.connectedSites.ignoredSites = groupedSites.get(UserChoice.Ignored) ?? [];
+        state.connectedSites.filter = search;
       });
     },
 
-    setFilter: (search?: string) => {
```

Comment:
`setFilter` was previously calculating a bunch of derived values and setting _them_ to state, which should really be done in selectors. So I simplified `setFilter` to just update the filter property.